src/main/scala/com/johnsnowlabs/ml/tensorflow/TensorResources.scala

```diff
@@ -12,24 +14,40 @@ class TensorResources {
   private val tensors = ArrayBuffer[Tensor[_]]()
 
   def createTensor[T](obj: T): Tensor[_] = {
-    val result = if (obj.isInstanceOf[String]) {
-      Tensor.create(obj.asInstanceOf[String].getBytes("UTF-8"), classOf[String])
-    }
-    else {
-      Tensor.create(obj)
+    val result = obj match {
+      case str: String =>
+        Tensor.create(str.getBytes("UTF-8"), classOf[String])
```

Comment:
Is this a style improvement or is the actual casting to string expensive?

Reply:
It is recommended to use case instead of `obj.isInstanceOf[String]`